### PR TITLE
Externalize credentials - Mcp-1816

### PR DESCRIPTION
**Pull request by wlin2019**
Using environment variables for username and password, and updating the project

## Description

1. Replace the credentials with environment variables. MCP-1816
2. Remove opa subproject as it is not used.3. 3. MCP-1858

The code change is based the code in branch MCP-1816 and MCP-1819 that Dimitri worked on. Since the current develop has changed quite a bit, the code merge has problems. This code change (PR) leaves out two things that seem to be addressed in MCP-1819. 1) admin account for RabbitMQ. 2) Some Postgres/Flyway related setting. See comments in MCP-1816

### What was the problem?

For security, the credentials should not be hard coded. opa sub project has no use in this project.

### How does this fix it?

Use the environment variables for credentials, and remove unused code.

### Jira Tickets

MCP-1816, MCP-1819, MCP-1858

## How to test this PR

- Step 1: Get updated setevn.sh and private.pem from abd-vro-dev-secretes and copy them to the project build directory.
- Step 2: Run setenv.sh.
- Step 3: Follow the README to build the application.

## Reminders

- Review [Pull-Requests](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests) guidelines
- [ ] If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) and
[Roadmap](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Roadmap) wiki pages
- [ ] If changing software behavior, post a link to the PR in Slack channel #benefits-virtual-regional-office

### CODE REVIEW

**Review by au2 — CHANGES_REQUESTED** [review]
Please take a look at other places. One place we hit is application-compose.yml

File: app/src/docker/docker-compose.yml [hunk 1]
```diff
@@ -40,7 +25,6 @@ services:
     ports:
       - "5672:5672"
       - "15672:15672"
-    restart: on-failure:5
```
[the remark this answers is not in the document]
Comment (wlin2019) [reply app/src/docker/docker-compose.yml]: Good catch! They got merged out. Put them back now.

File: app/src/docker/docker-compose.yml [hunk 2]
```diff
@@ -101,10 +75,8 @@ services:
       JAVA_PROFILE: "-Dspring.profiles.include=compose"
       # TODO: move these RabbitMQ configurations into a reusable yaml block
       RABBITMQ_PLACEHOLDERS_HOST: rabbitmq1
-      RABBITMQ_PLACEHOLDERS_USERNAME: guest
-      RABBITMQ_PLACEHOLDERS_USERPASSWORD: guest
-    security_opt: 
-      - no-new-privileges:true
+      RABBITMQ_PLACEHOLDERS_USERNAME: ${RABBITMQ_USERNAME}
+      RABBITMQ_PLACEHOLDERS_USERPASSWORD: ${RABBITMQ_PASSWORD}
```
[the remark this answers is not in the document]
Comment (wlin2019) [reply app/src/docker/docker-compose.yml]: Changed the names, and updated in setenv.sh in secrets repo as well.

File: app/src/docker/docker-compose.yml [hunk 3]
```diff
@@ -51,16 +35,15 @@ services:
...
     hostname: vro-api-postgres-service
     expose:
       - "5432"
     environment:
-      POSTGRES_USER: vro_user
-      POSTGRES_PASSWORD: not-the-password
+      POSTGRES_USER: ${POSTGRES_USER}
+      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
       POSTGRES_DB: vro
     volumes:
```
[the remark this answers is not in the document]
Comment (wlin2019) [reply app/src/docker/docker-compose.yml:50]: I'm reluctant to do this as "vro" is also used in url. But anyway, I created an env var, POSTGRES_DBNAME, for it (It is also used as POSTGRES_DBNAME in the project).

File: app/src/main/resources/application-compose.yml [hunk 4]
```diff
@@ -3,12 +3,12 @@
 spring:
   datasource:
     url: jdbc:postgresql://vro-api-postgres-service:5432/vro
-    username: vro_user
-    password: not-the-password
+    username: ${POSTGRES_USER}
+    password: ${POSTGRES_PASSWORD}
     hikari:
       jdbc-url: jdbc:postgresql://vro-api-postgres-service:5432/vro
```
Comment (au2) [inline app/src/main/resources/application-compose.yml:9]: Please put the url in an environment variable here and add it to docker-compose. Maybe POSTGRES_URL.

**Review by au2 — CHANGES_REQUESTED** [review]
Added one more inline comment.

**Review by au2 — CHANGES_REQUESTED** [review]
Please add 
    environment:
      RABBITMQ_DEFAULT_USER: ${RABBITMQ_PLACEHOLDERS_USERNAME}
      RABBITMQ_DEFAULT_PASS: ${RABBITMQ_PLACEHOLDERS_USERPASSWORD}
to docker-compose. This will give us a quick ability to change default rabbit user.

**Pull request merged**